confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java

```diff
@@ -331,6 +348,121 @@ public PermissionInfoResp executeListUserPrivileges(AuthorReq plan) throws AuthE
     }
   }
 
+  @Override
+  public boolean processTakeSnapshot(File snapshotDir) throws TException, IOException {
+    SystemFileFactory systemFileFactory = SystemFileFactory.INSTANCE;
+    File userFolder = systemFileFactory.getFile(commonConfig.getUserFolder());
+    File userSnapshotDir = systemFileFactory.getFile(snapshotDir, userSnapshotFileName);
+    File userTmpSnapshotDir =
+        systemFileFactory.getFile(userSnapshotDir.getAbsolutePath() + "-" + UUID.randomUUID());
+    File roleFolder = systemFileFactory.getFile(commonConfig.getRoleFolder());
+    File roleSnapshotDir = systemFileFactory.getFile(snapshotDir, roleSnapshotFileName);
+    File roleTmpSnapshotDir =
+        systemFileFactory.getFile(roleSnapshotDir.getAbsolutePath() + "-" + UUID.randomUUID());
+
+    boolean result = true;
+    synchronized (IAuthorizer.class) {
+      try {
```

Comment:
If some other authorInfo write operation doesn't synchronize its process, will this synchronize mechanism work?

Reply:
It seems this synchronize is only used for avoid concurrent snapshot creation. The snapshot invoking program should take the responsibility to acquire the global write lock of configNode.

Reply:
I use ReentrantReadWriteLock in AuthorizerManager to fix this problem.